Argo/Operators/DecodeDecoded.swift

```diff
@@ -1,43 +1,152 @@
+             decode operation
+*/
+public func <| <T where T: Decodable, T == T.DecodedType>(json: JSON, key: String) -> Decoded<T> {
   return json <| [key]
 }
 
-// Pull optional value from JSON
-public func <|? <A where A: Decodable, A == A.DecodedType>(json: JSON, key: String) -> Decoded<A?> {
+/**
+  Attempt to decode an optional value at the specified key into the requested
+  type
+
+  This operator is used to decode an optional value from the JSON. If the key
+  isn't present in the JSON, this will still return `Success`. However, if the
+  key exists but the object assigned to that key is unable to be decoded into the
+  requested type, this will return `Failure`.
+
+  - parameter json: The `JSON` object containing the key
+  - parameter key: The key for the object to decode
+
+  - returns: A `Decoded` optional value representing the success or failure of the
+             decode operation
+*/
+public func <|? <T where T: Decodable, T == T.DecodedType>(json: JSON, key: String) -> Decoded<T?> {
   return .optional(json <| [key])
 }
 
-// Pull embedded value from JSON
-public func <| <A where A: Decodable, A == A.DecodedType>(json: JSON, keys: [String]) -> Decoded<A> {
-  return flatReduce(keys, initial: json, combine: decodedJSON) >>- A.decode
+/**
+  Attempt to decode a value at the specified key path into the requested type
+
+  This operator is used to decode a mandatory value from the JSON. If the
+  decoding fails for any reason, this will result in a `Failure` being
+  returned.
+
+  - parameter json: The `JSON` object containing the key
+  - parameter keys: The key path for the object to decode, represented by an
+                    array of strings
+
+  - returns: A `Decoded` value representing the success or failure of the
+             decode operation
+*/
+public func <| <T where T: Decodable, T == T.DecodedType>(json: JSON, keys: [String]) -> Decoded<T> {
+  return flatReduce(keys, initial: json, combine: decodedJSON) >>- T.decode
 }
 
-// Pull embedded optional value from JSON
-public func <|? <A where A: Decodable, A == A.DecodedType>(json: JSON, keys: [String]) -> Decoded<A?> {
+/**
+  Attempt to decode an optional value at the specified key path into the
+  requested type
+
+  This operator is used to decode an optional value from the JSON. If any of
+  the keys in the key path aren't present in the JSON, this will still return
+  `Success`. However, if the key path exists but the object assigned to the final
+  key is unable to be decoded into the requested type, this will return
+  `Failure`.
+
+  - parameter json: The `JSON` object containing the key
+  - parameter keys: The key path for the object to decode, represented by an
+                    array of strings
+
+  - returns: A `Decoded` optional value representing the success or failure of the
+             decode operation
+*/
+public func <|? <T where T: Decodable, T == T.DecodedType>(json: JSON, keys: [String]) -> Decoded<T?> {
```

Comment:
Line Length Violation: Line should be 100 characters or less: currently 103 characters (line_length)